src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py

```diff
@@ -661,10 +661,39 @@ def _validator(namespace):
     return _validator
 
 
-def process_nw_flow_log_set_namespace(namespace):
+def process_nw_test_connectivity_namespace(namespace):
 
     from azure.cli.core.commands.arm import parse_resource_id
 
+    compute_client = get_mgmt_service_client(ResourceType.MGMT_COMPUTE).virtual_machines
+    vm_name = parse_resource_id(namespace.source_resource)['name']
+    rg = namespace.resource_group_name or \
+        parse_resource_id(namespace.source_resource).get('resource_group', None)
+    if not rg:
+        raise CLIError('usage error: --source-resource ID | '
+                       '--source-resource NAME --resource-group NAME')
+    vm = compute_client.get(rg, vm_name)
+    namespace.location = vm.location  # pylint: disable=no-member
+    get_network_watcher_from_location(remove=True)(namespace)
```

Comment:
I found it a bit odd to use a `get` function to remove a property from the namespace.